src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -8,146 +8,326 @@
-                    account.setContinuationAccountNumber(continuationAccountNumber);
-                }
-    
-                // INCOME STREAM CHART OF ACCOUNTS CODE
-                if (StringUtils.isNotBlank(incomeStreamFinancialCoaCode)) {
-                    account.setIncomeStreamFinancialCoaCode(incomeStreamFinancialCoaCode);
-                }
-    
-                // INCOME STREAM ACCOUNT NUMBER
-                if (StringUtils.isNotBlank(incomeStreamAccountNumber)) {
-                    account.setIncomeStreamAccountNumber(incomeStreamAccountNumber);
-                }
-    
-                // CG CATL FED DOMESTIC ASSIST NBR
-                if (StringUtils.isNotBlank(accountCfdaNumber)) {
-                    account.setAccountCfdaNumber(accountCfdaNumber);
-                }
-    
-                // FINANCIAL HIGHER ED FUNCTION CODE
-                if (StringUtils.isNotBlank(financialHigherEdFunctionCd)) {
-                    account.setFinancialHigherEdFunctionCd(financialHigherEdFunctionCd);
-                }
-    
-                // SUFFICIENT FUNDS CODE
-                if (StringUtils.isNotBlank(accountSufficientFundsCode)) {
-                    account.setAccountSufficientFundsCode(accountSufficientFundsCode);
-                }
-    
-                // LABOR BENEFIT RATE CATEGORY CODE
-                if (StringUtils.isNotBlank(getLaborBenefitRateCategoryCode())) {
-                    account.setLaborBenefitRateCategoryCode(getLaborBenefitRateCategoryCode());
-                }
-    
-                // PENDING ACCOUNT SUFFICIENT FUNDS CODE INDICATOR
-                if (pendingAcctSufficientFundsIndicator != null) {
-                    account.setPendingAcctSufficientFundsIndicator(pendingAcctSufficientFundsIndicator);
-                }
-    
-                // MAJOR REPORTING CATEGORY CODE 
-                if (StringUtils.isNotBlank(majorReportingCategoryCode)) {
-                    ((AccountExtendedAttribute) account.getExtension()).setMajorReportingCategoryCode(majorReportingCategoryCode);
-                }
-                
-                
-                persistables.add(account);
-    
-            }
-        }
 
+        for (AccountGlobalDetail accountGlobalDetail : accountGlobalDetails) {
+        	updateAccountValuesAndAddToPersistablesList(persistables, accountGlobalDetail);
+        }
+        
         return persistables;
     }
+
+	private void updateAccountValuesAndAddToPersistablesList(List<PersistableBusinessObject> persistables,AccountGlobalDetail accountGlobalDetail) {
+		Account account = updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(accountGlobalDetail);
+		
+		if (ObjectUtils.isNotNull(account)){
+			persistables.add(account);   
+		}
+	}
     
-    public String getMajorReportingCategoryCode() {
+    private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(GlobalBusinessObjectDetailBase globalDetail) {
```

Comment:
Please update the horizontal formatting of this method; its current arrangement makes it harder to spot the start of the subsequent method.